src/NodeAnalyzer/ValueResolver.php

```diff
@@ -4,33 +4,19 @@
 
 namespace Bladestan\NodeAnalyzer;
 
-use PhpParser\ConstExprEvaluationException;
-use PhpParser\ConstExprEvaluator;
 use PhpParser\Node\Expr;
 use PHPStan\Analyser\Scope;
-use PHPStan\Type\ConstantScalarType;
 
 final class ValueResolver
 {
-    public function __construct(
-        private readonly ConstExprEvaluator $constExprEvaluator
-    ) {
-    }
-
     public function resolve(Expr $expr, Scope $scope): mixed
     {
-        try {
-            return $this->constExprEvaluator->evaluateDirectly($expr);
-        } catch (ConstExprEvaluationException) {
-        }
-
         $exprType = $scope->getType($expr);
-
-        /** @phpstan-ignore phpstanApi.instanceofType */
-        if ($exprType instanceof ConstantScalarType) {
-            return $exprType->getValue();
+        $constantScalarValues = $exprType->getConstantScalarValues();
+        if (count($constantScalarValues) !== 1) {
```

Comment:
I was a bit scared the make these assumptions about the number of elements returned :)

Reply:
You could also fix it for multiple values. They typically occur in a union. So `'foo'|'bar'`. But I'm not sure what to do with it in the rest of the code.

Reply:
I think we will cross that bridge when we have to :)